wordpress-mobile/WordPress-iOS
# Reader Comments: Implement functionality for the new Follow button

Refs `pctCYC-9x-p2`

This adds the implementation for the new Follow button, mainly the flow that introduces the user to enable notifications through the little snackbar views. Some notes on the changes:

- Added a method on `UIViewController+Notice` so that I could show a notice with an action handler from Objective-C. I know this is slightly unrelated to the PR, but since it's a minor change and required by this PR, I figured to just include it together.
- The bottom sheet implementation will be introduced in the next PR. So it's expected that tapping the bell button does nothing for now.

Designs:

<img width="298" alt="Screen Shot 2021-10-20 at 02 53 26" src="https://user-images.githubusercontent.com/1299411/137980944-7e52c19f-1ca2-4ecf-8a50-d94e4ff96531.png">

## To Test

Prepare some posts that are unsubscribed, and ensure that the `Follow Conversation via Notifications` flag is enabled.

- Go to Reader, and select any unsubscribed post.
- Tap on the Follow button.
- Verify that the notice appears, and displays the content correctly with the "Enable" action button.
- Verify that the Follow button now turns into a bell icon.
- Tap on the "Enable" button.
- Verify that the notice appears, and displays the content correctly with the "Undo" action button.
- Tap on the "Undo" button.
- Verify that the notice appears informing that the Undo process has been completed.

## Regression Notes
1. Potential unintended areas of impact
The feature is still hidden behind a flag, but the current follow functionality may be impacted. Additionally, this introduced some additional methods to `UIViewController+Notice` and _could_ disrupt the existing usages of `displayNotice(title: String, message: String?)` due to it being slightly modified.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested and ensured that both implementations work as expected.

3. What automated tests I added (or what prevented me from doing so)
n/a

PR submission checklist:
